tests/unit/conftest.py

```diff
@@ -0,0 +1,40 @@
+"""tests/unit/conftest.py: pytest fixtures for unit-testing functions in the xen-bugtool script"""
+import os
+
+import pytest
+
+
+@pytest.fixture(scope="session")
+def testdir():
+    """Test fixture to get the directory of the unit test for finding other files relative to it"""
+    return os.path.dirname(__file__)
+
+
+@pytest.fixture(scope="session")
+def bugtool(testdir):
+    """Test fixture to import the xen-bugtool script as a module for executing unit tests on functions"""
+
```

Comment:
I notice extra empty lines and overlong lines. Is there a convention for Python? Overlong lines make reviews more difficult so I would avoid them.

Reply:
Thanks for the review!

There is no convention on the max-column length in Python. Different project use different widths.

The formatting of the files in the tests/ directory is now governed by Python's formatter `black`, which is a widely adopted, opinionated (enforcing) Python formatter.

It enforces the amount of empty lines around functions, imports and so on, and after getting a little bit used to it, it helps a lot because of the consistency of formatting and the freedom of thought, as black fully takes over all aspects of formatting.

I also prefer to not allow long lines beyond a certain limit and prefer shorter lines of course.

Traditionally, XenServer Python Project did not enforce a maximum line length, there are very long lines in them in the xen-bugtool script and xsconsole.

For now, I've setup pre-commit checkers and GitHub CI to enforce a maximum line lengh, but I agree that I should aim at shortening it.

In the meantime, I'll shorten or break my comments/code into narrower blocks.

I'll look at reducing the checked maximum text width, thanks for sharing your feedback!